bin/gaia_marionette_retry.js

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env node
+var Promise = require('es6-promise').Promise;
+var format = require('util').format;
+var path = require('path');
+var spawn = require('child_process').spawn;
+
+function main() {
+  // 0 => node
+  // 1 => ./bin/gaia_marionette_retry.js
+  process.argv.splice(0, 2);
```

Comment:
Everything from here to the `runTests` invocation is just parsing whatever usually gets passed to `./bin/gaia-marionette` so that we can separate the test files that need to be run from all of the various args